Excercise_2.java

```diff
@@ -0,0 +1,129 @@
+ class StackAsLinkedList {
+
+    StackNode root;
+    static class StackNode {
+        int data;
+        StackNode next;
+
+        StackNode(int data)
+        {
+            this.data=data;
+            this.next=null;
+            //Constructor here
+        }
+
+
+    }
+
+public void printElements(){
+    System.out.println("elements ccurrently ");
+    StackNode first=root;
+    while(root!=null){
+        System.out.println(root.data);
+        root=root.next;
+    }
+    root=first;
+}
+    public boolean isEmpty()
+    {
+        if(root==null){
+            return true;
+        }
+        return false;
+        //Write your code here for the condition if stack is empty.
+    }
+
+    public void push(int data)
+    {
+        StackNode first=root;
+        //Write code to push data to the stack.
+//        StackNode first=root;
+     if(root==null){
+         root=new StackNode(data);
+     }else{
+         if(root.next!=null){
+         while(root.next!=null){
```

Comment:
instead of iterating to find the tail, maintain a separate tail class variable to reduce the time to O(1)